src/ponyfills/objects.js

```diff
@@ -0,0 +1,176 @@
+import deepmerge from 'deepmerge';
+
+export const deepMerge = deepmerge;
+export function deepClone(obj) {
+  return JSON.parse(JSON.stringify(obj));
+}
+
+/**
+ * Check if a variable is an Object. This function considers Null and Array to not be Objects.
+ *
+ * @param variable The variable that is to tested
+ * @returns {boolean} True if the variable is an object
+ */
+export function isObject(variable) {
+  return typeof variable === 'object' && variable !== null && !Array.isArray(variable);
+}
+
+/**
+ * Convert a hierarchical map into a flattened map
+ *
+ * @param {Object} map A map with hierarchical keys
+ * @returns {Object} A map with hierarchical keys flattened
+ */
+export function flatten(map, filter) {
+  function recurse(current, parent_key) {
+    let items = {};
+    Object.keys(current).filter(filter || function() { return true; }).forEach(function(k) {
+      let v = current[k];
+      let new_key = parent_key ? (parent_key + '.' + k) : k;
+      if (isObject(v)) {
+        items = assign(items, recurse(current[k], new_key));
+      } else {
+        items[new_key] = v;
+      }
+    });
+
+    return items;
+  }
+
+  return recurse(map, '');
+}
+
+export function flattenKeys(map, filter) {
+  function recurse(current, parent_key) {
+    let items = [];
+    Object.keys(current).filter(filter || function() { return true; }).forEach(function(k) {
+      let v = current[k];
+      let new_key = parent_key ? (parent_key + '.' + k) : k;
+      items.push(new_key);
+      if (isObject(v)) {
+        items = items.concat(recurse(current[k], new_key));
+      }
+    });
+
+    return items;
+  }
+
+  return recurse(map, '');
+}
+
+export function removeValueForKey(key, map) {
+  function recurse(keys, index, map) {
+    let key = keys[index];
+    if (index === (keys.length - 1)) {
+      delete map[key];
+      return true;
+    }
+
+    if (!(key in map)) {
+      return false;
+    }
+
+    if (recurse(keys, index + 1, map[key]) && Object.keys(map[key]).length === 0) {
+      delete map[key];
+      return true;
+    }
+  }
+
+  return recurse(key.split('\.'), 0, map);
```

Comment:
Curious: why the backslash in the pattern `\.`? Is a residual from originally using a regex? I tested it and it works properly. I'm just wondering if it was deliberate.

Reply:
Yep, just habit. I'll remove it.